v2/tools/generator/internal/astmodel/code_generation_context.go

```diff
@@ -83,34 +85,62 @@ func (codeGenContext *CodeGenerationContext) MustGetImportedPackageName(referenc
 }
 
 // GetGeneratedPackage gets a reference to the PackageDefinition referred to by the provided reference
-func (codeGenContext *CodeGenerationContext) GetGeneratedPackage(reference PackageReference) (*PackageDefinition, error) {
+func (ctx *CodeGenerationContext) GetGeneratedPackage(reference PackageReference) (*PackageDefinition, error) {
 	// Make sure that we're actually importing that package -- don't want to allow references to things we aren't importing
-	_, err := codeGenContext.GetImportedPackageName(reference)
-	if !reference.Equals(codeGenContext.currentPackage) && err != nil {
+	_, err := ctx.GetImportedPackageName(reference)
+	if !reference.Equals(ctx.currentPackage) && err != nil {
 		return nil, err
 	}
 
-	packageDef, ok := codeGenContext.generatedPackages[reference]
+	packageDef, ok := ctx.generatedPackages[reference]
 	if !ok {
 		return nil, errors.Errorf("%s not imported", reference)
 	}
 	return packageDef, nil
 }
 
-// GetImportedDefinition looks up a particular type definition in a package available in this context
-func (codeGenContext *CodeGenerationContext) GetImportedDefinition(typeName TypeName) (TypeDefinition, error) {
-	pkg, err := codeGenContext.GetGeneratedPackage(typeName.PackageReference)
+// GetOrErr looks up a particular type definition in a package available in this context
+func (ctx *CodeGenerationContext) GetOrErr(name TypeName) (TypeDefinition, error) {
```

Comment:
I strongly dislike the name `GetOrErr()` - we don't use `OrErr` as a suffix on any other methods. 

What was wrong with the previous name `GetImportedDefinition()`? If it must be renamed, consider taking inspiration from the comment: `LookupDefinition()`.

Reply:
> What was wrong with the previous name `GetImportedDefinition()`?

The issue is that since this is an interface `ReadonlyTypes`, that not all collections of `ReadonlyTypes` have a notion of an "imported definition" - for example `TypeDefinitionSet` doesn't really. It has definitions but no notion of "imported vs not imported"

What about: 

`GetDefinition (TypeDefinition, error)`
`MustGetDefinition TypeDefinition`

and I just delete `TryGet` entirely since you can do the same thing with `GetDefinition` and checking the error and it has very minimal usage (2 calls).

Reply:
Did the above (as discussed over Teams)